src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -456,35 +456,35 @@ public SolutionCompilationState AddAnalyzerReferences(StateChange stateChange, I
             arg: analyzerReferences);
     }
 
-    public SolutionCompilationState AddAnalyzerReferences(SolutionState newSolutionState)
+    public SolutionCompilationState AddAnalyzerReferences(IReadOnlyCollection<AnalyzerReference> analyzerReferences)
     {
         // Note: This is the codepath for adding analyzers from vsixes.  Importantly, we do not ever get SGs added
         // from this codepath, and as such we do not need to update the compilation trackers.  The methods that add SGs
         // all come from entrypoints that are specific to a particular project.
-        return Branch(newSolutionState);
+        return Branch(this.Solution.AddAnalyzerReferences(analyzerReferences));
     }
 
-    public SolutionCompilationState RemoveAnalyzerReference(SolutionState newSolutionState)
+    public SolutionCompilationState RemoveAnalyzerReference(AnalyzerReference analyzerReference)
     {
         // Note: This is the codepath for adding analyzers from vsixes.  Importantly, we do not ever get SGs added
         // from this codepath, and as such we do not need to update the compilation trackers.  The methods that add SGs
         // all come from entrypoints that are specific to a particular project.
-        return Branch(newSolutionState);
+        return Branch(this.Solution.RemoveAnalyzerReference(analyzerReference));
```

Comment:
>RemoveAnalyzerReference [](http://example.com/codeflow?start=36&length=23)

nit: comments in these three methods seem like they should have some difference

Reply:
yup.  fixed.